src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -208,16 +208,46 @@ class RefreshIndexTests extends QueryTest with HyperspaceSuite {
 
       // Replace a source data file with a new file with same name but different properties.
       val deletedFile = deleteOneDataFile(nonPartitionedDataPath)
-      FileUtils.createFile(
-        deletedFile.getFileSystem(new Configuration),
-        deletedFile,
-        "I am some random content :).")
-
-      val ex = intercept[HyperspaceException](
-        hyperspace.refreshIndex(indexConfig.indexName, REFRESH_MODE_INCREMENTAL))
+      val sourcePath = new Path(spark.read.parquet(nonPartitionedDataPath).inputFiles.head)
+      val fs = deletedFile.getFileSystem(new Configuration)
+      fs.copyToLocalFile(sourcePath, deletedFile)
+      val prevIndexLogEntry = getLatestStableLog(indexConfig.indexName)
+      assert(logManager(systemPath, indexConfig.indexName).getLatestId().get == 1)
       assert(
-        ex.getMessage.contains("Index refresh (to handle deleted source data) aborted. " +
-          "Existing source data file info is changed"))
+        prevIndexLogEntry.content.fileInfos
+          .forall(_.name.contains(s"${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=0")))
+
+      val indexPath = PathUtils.makeAbsolute(s"$systemPath/${indexConfig.indexName}")
+      new RefreshDeleteAction(
+        spark,
+        IndexLogManagerFactoryImpl.create(indexPath),
+        IndexDataManagerFactoryImpl.create(indexPath))
+        .run()
+
+      val indexLogEntryAfterDeleteRefresh = getLatestStableLog(indexConfig.indexName)
+      assert(logManager(systemPath, indexConfig.indexName).getLatestId().get == 3)
+      assert(indexLogEntryAfterDeleteRefresh.deletedFiles.isEmpty)
+      assert(indexLogEntryAfterDeleteRefresh.appendedFiles.size == 1)
```

Comment:
According to the PR description below, the modified file should be recorded as "deletedFiles"?
```
RefreshDeleteAction handles the modified files as deleted files, and also as appended files.
```

Reply:
Ah, nm, I think we need the test coverage for `RefreshAppendAction` where `deletedFiles` become non-empty.

Reply:
@sezruby can you do a follow up PR to add the coverage? (I manually verified by updating the test slightly to unblock the current release.)

Reply:
Created #213.